server/src/internalClusterTest/java/org/elasticsearch/reservedstate/service/FileSettingsServiceIT.java

```diff
@@ -498,6 +500,47 @@ public void testSettingsAppliedOnMasterReElection() throws Exception {
         assertClusterStateSaveOK(savedClusterState.v1(), savedClusterState.v2(), "43mb");
     }
 
+    public void testHealthIndicator() throws Exception {
+        internalCluster().setBootstrapMasterNodeIndex(0);
+        logger.info("--> start a data node to act as the health node");
+        internalCluster().startNode(Settings.builder().put(dataOnlyNode()).put("discovery.initial_state_timeout", "1s"));
+
+        logger.info("--> start master node");
+        final String masterNode = internalCluster().startMasterOnlyNode(
+            Settings.builder().put(INITIAL_STATE_TIMEOUT_SETTING.getKey(), "0s").build()
+        );
+        FileSettingsService masterFileSettingsService = internalCluster().getInstance(FileSettingsService.class, masterNode);
+        assertBusy(() -> assertTrue(masterFileSettingsService.watching()));
+
+        ensureStableCluster(2);
+
+        var healthNode = HealthNode.findHealthNode(clusterService().state()).getName();
+
+        // Initially, all is well
+        assertBusy(() -> {
+            FetchHealthInfoCacheAction.Response healthNodeResponse = client(healthNode).execute(
```

Comment:
The `FetchHealthInfoCacheAction.TransportAction` is of type `TransportHealthNodeAction` so it will automatically get routed to the health node, meaning we don't need to specify a specific client here.